src/Centreon/Application/Controller/MonitoringResourceController.php

```diff
@@ -340,47 +284,234 @@ public function detailsService(int $hostId, int $serviceId): View
         $this->denyAccessUnlessGrantedForApiRealtime();
 
         /**
-         * @var Service $service
+         * @var Contact $contact
          */
-        $service = $this->monitoring
-            ->filterByContact($this->getUser())
-            ->findOneService($hostId, $serviceId);
+        $contact = $this->getUser();
+
+        $filter = (new ResourceFilter())
+            ->setTypes([ResourceFilter::TYPE_SERVICE])
+            ->setHostIds([$hostId])
+            ->setServiceIds([$serviceId]);
+
+        $resources = $this->resource
+            ->filterByContact($contact)
+            ->findResources($filter);
+
+        if (empty($resources)) {
+            return View::create(null, Response::HTTP_NOT_FOUND, []);
+        }
+
+        $resource = $resources[0];
+
+        $this->providePerformanceGraphEndpoint([$resource]);
+        $this->provideLinks($resource, $contact);
+
+        $this->resource->enrichServiceWithDetails($resource);
+
         try {
+            $service = (new Service())
+                ->setId($resource->getId())
+                ->setHost(
+                    (new Host())
+                        ->setId($resource->getParent()->getId())
+                )
+                ->setCommandLine($resource->getCommandLine());
             $this->monitoring->hidePasswordInCommandLine($service);
+            $resource->setCommandLine($service->getCommandLine());
         } catch (\Throwable $ex) {
-            $service->setCommandLine(
+            $resource->setCommandLine(
                 sprintf('Unable to hide passwords in command (Reason: %s)', $ex->getMessage())
             );
         }
 
-
-        if ($service === null) {
-            return View::create(null, Response::HTTP_NOT_FOUND, []);
-        }
-
         $context = (new Context())
-            ->setGroups(array_merge([
-                ResourceDetailsService::SERIALIZER_GROUP_DETAILS,
-                ResourceStatus::SERIALIZER_GROUP_MAIN,
-                Acknowledgement::SERIALIZER_GROUP_FULL,
-            ], Downtime::SERIALIZER_GROUPS_SERVICE))
+            ->setGroups(array_merge(
+                static::SERIALIZER_GROUPS_LISTING,
+                [ResourceEntity::SERIALIZER_GROUP_DETAILS, Acknowledgement::SERIALIZER_GROUP_FULL],
+                Downtime::SERIALIZER_GROUPS_SERVICE
+            ))
             ->enableMaxDepth();
 
+        $context->addExclusionStrategy(new ResourceExclusionStrategy());
+
         return $this
-            ->view($this->resource->enrichServiceWithDetails($service))
+            ->view($resource)
             ->setContext($context);
     }
 
+    /**
+     * Add performance graph endpoint on resources which have performance data
+     *
+     * @param ResourceEntity[] $resources
+     * @return void
+     */
+    private function providePerformanceGraphEndpoint($resources)
```

Comment:
The name of this function does not match its description.
provide performance != Add performance

Missing variable typing